disco_aws_automation/disco_aws.py

```diff
@@ -715,13 +726,19 @@ def spinup(self, hostclass_dicts, stage=None, no_smoke=False, testing=False, cre
                 for (hostclass, termination_policies, hdict) in hostclass_iter]
 
             if metadata[0]:
+                for _hc in metadata:
+                    if _hc.get('spotinst'):
```

Comment:
Can the  call to `self.elastigroup.wait_for_instance_id` be part of `wait_for_autoscaling_instances`